crates/llm-remote-http/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+use anyhow::Result;
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    Client, Url,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use spin_core::async_trait;
+use spin_llm::LlmEngine;
+use spin_world::llm::{self as wasi_llm, InferencingParams};
+use std::collections::HashSet;
+
+#[derive(Clone)]
+pub struct RemoteHttpLlmEngine {
+    auth_token: String,
+    url: Url,
+    client: Option<Client>,
+    allowed_models: HashSet<String>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all(serialize = "camelCase"))]
```

Comment:
The PR description had some of the fields looking a bit non-camel-cased e.g. `repeatPenalty_last_n_token_count` or `PromptTokenCount`.  I assume from the serde annotations that those were typos and that the interface is consistent?

Reply:
Yea - those are typos. The JSON response is `camelCased`.